WordPress/Classes/ViewRelated/Menus/MenuItemEditingViewController.m

```diff
@@ -0,0 +1,470 @@
+@property (nonatomic, strong) IBOutlet NSLayoutConstraint *footerViewHeightConstraint;
+
+@property (nonatomic, assign) MenuItemEditingViewControllerContentLayout contentLayout;
+@property (nonatomic, strong) NSArray *layoutConstraintsForDisplayingTypeView;
+@property (nonatomic, strong) NSArray *layoutConstraintsForDisplayingSourceView;
+@property (nonatomic, strong) NSArray *layoutConstraintsForDisplayingSourceAndTypeViews;
+
+@property (nonatomic, assign) BOOL observesKeyboardChanges;
+@property (nonatomic, assign) BOOL sourceViewIsTyping;
+
+@end
+
+@implementation MenuItemEditingViewController
+
+- (void)dealloc
+{
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (id)initWithItem:(MenuItem *)item blog:(Blog *)blog
+{
+    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"MenuItemEditing" bundle:nil];
+    self = [storyboard instantiateInitialViewController];
+    if (self) {
+        
+        _blog = blog;
+        
+        // Keep track of changes to the item on a scratch contect and scratch item.
+        NSManagedObjectID *itemObjectID = item.objectID;
+        NSManagedObjectContext *scratchContext = [[NSManagedObjectContext alloc] initWithConcurrencyType:NSMainQueueConcurrencyType];
+        scratchContext.parentContext = blog.managedObjectContext;
+        scratchContext.mergePolicy = NSMergeByPropertyObjectTrumpMergePolicy;
+        _scratchObjectContext = scratchContext;
+        
+        [scratchContext performBlockAndWait:^{
+            NSError *error;
+            MenuItem *itemInContext = [scratchContext existingObjectWithID:itemObjectID error:&error];
+            if (error) {
+                DDLogError(@"Error occurred obtaining existing MenuItem object in context: %@", error);
+            }
+            _item = itemInContext;
+        }];
+    }
+    
+    return self;
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    self.automaticallyAdjustsScrollViewInsets = NO;
+    self.view.backgroundColor = [UIColor whiteColor];
+    
+    self.stackView.translatesAutoresizingMaskIntoConstraints = NO;
+    
+    self.headerView.item = self.item;
+    
+    self.headerView.delegate = self;
+    self.typeViewController.delegate = self;
+    self.sourceViewController.delegate = self;
+    self.footerView.delegate = self;
+    
+    [self.stackView bringSubviewToFront:self.headerView];
+    
+    [self loadContentLayoutConstraints];
+    [self updateLayoutIfNeeded];
+    
+    self.typeViewController.selectedItemType = self.item.type;
+    [self.typeViewController loadPostTypesForBlog:self.blog];
+    
+    self.sourceViewController.blog = self.blog;
+    self.sourceViewController.item = self.item;
+}
+
+- (void)viewDidLayoutSubviews
+{
+    [super viewDidLayoutSubviews];
+    
+}
+
+- (void)viewWillAppear:(BOOL)animated
+{
+    [super viewWillAppear:animated];
+    
+    if (self.modalPresentationStyle == UIModalPresentationFormSheet) {
```

Comment:
Looks like this isn't needed anymore?  I'm not seeing where this VC is ever presented as a form sheet.